discussions/prog_repo.md

```diff
@@ -0,0 +1,192 @@
+
+# What should a future version look like
+
+This section examines simple ways to create a location where eBPF Programs from trusted parties
+can be uploaded and made available for others to download. This can also enable users to provide
+rating, reviews, and tags to the packages.
+
+### Build Process
+
+Contributed eBPF programs can be built on common images for Linux and other platforms.
+Contributors are expected to build their code using the most recent image used by our build system.
+Contributors could have a choice to build on Linux only or other platforms.
+
+Build artifacts (i.e., eBPF program bytecode and user space binaries) could be stored
+in a public file storage repository. Users could download the artifacts directly
+from this repository. In fact, such a repository could also be considered the
+eBPF Package Repository, and the source repository could be a separate entity.
+
+### Portability
+
+It is hard to discuss a build system without broaching the topic of portability.
+The eBPF portability story has improved recently with the introduction of eBPF CO-RE on Linux.
+Previous issues with portability and the eBPF CO-RE solution are explained here:
+
+https://nakryiko.com/posts/bpf-portability-and-co-re/
+
+Because an eBPF Package Repository is a place from where contributed eBPF programs (byte code)
+can be downloaded to run on a variety of kernel versions, we propose that the repository follows
+best practices for compatibility, such as using eBPF CO-RE for Linux. Similar best practices
+can be followed on non-Linux platforms as they mature and become available.
+The user space components of eBPF programs pose a separate, complicated portability obstacle,
+which is compounded by the desire of the L3AF project to support user space programs in multiple
+languages. For the initial version of the repository, it should be sufficient to build
+(of a compiled language) and unit test for the x86_64 platform. The user space component
+should document any installation dependencies it has (e.g., MySQL, Grafana, Python libraries, etc.).
+Contributors to the repository would be responsible to provide the necessary build scripts
+and configuration.
+
+### Multiple Versions
+
+Regardless of whether the package is built from source or not, eBPF programs should use semantic
+versioning. The repository should then host previous versions (in addition to the current version)
+in some reasonable manner.
+
+### Automated Reviews
+
+There can be a mechanism to review source code, using tools to detect like code formatting errors,
+complexity, code issues, and code duplication. Similarly, vulnerability detection
+scanners can be used for artifacts before uploading.
+
+## Alternative to Hosting and Building Source Code
+
+An alternative to the L3AF Project hosting and building contributed source code would be for
+contributors to submit or self-host a signed package containing their eBPF programs and any
```

Comment:
Doesn't line 61 say this alternative is what would be the initial version?  In which case it's not an "alternative" per se?  Or am I misunderstanding the earlier section?

Reply:
@dthaler Initial version of eBPF Package Repository is to host eBPF Program source code in a github repo. But this section describes if someone does not want to share the source code, and still wants to share the eBPF Package, then it should be supported.

Reply:
In that case you'd probably want a more specific name for the particular eBPF Package Repository (e.g., "L3AF Public Package Repository") with that policy, since that may just discourage some from using that repository and instead use their own.  The document already allows for that, but I would prefer to separate the concept of package repositories in general, from the policies around a very specific one that may or may not be used, up to the admin of any particular l3afd installation.

Reply:
This is just an alternative hosting approach, we can decide on this in future on need basis.

Reply:
This issue will be addressed here https://github.com/l3af-project/eBPF-Package-Repository/issues/2